component/mcsd/updater.go

```diff
@@ -48,31 +50,80 @@ func buildUpdateTransaction(ctx context.Context, tx *fhir.Bundle, entries []fhir
 		}
 		log.Ctx(ctx).Debug().Msgf("Adding entry #%d to transaction: %s", i, resourceType)
 
-		setResourceMetaSource(resource, "")
+		updateResourceMeta(resource, *entry.FullUrl)
 		// Get or create local reference
-		// TODO: Lookup ID local to local mCSD Directory, not remote
-		//localResourceID := localRefMap[*entry.FullUrl]
-		//if localResourceID == "" {
-		//	localResourceID = fmt.Sprintf("urn:uuid:%s", uuid.NewString())
-		//	localRefMap[*entry.FullUrl] = localResourceID
-		//}
-		//resource["id"] = localResourceID
+		// TODO: If the resource already exists, we should look up the existing resource's ID
+		// TODO: We should scope resource IDs to the source (e.g. by prefixing with the source URL or a hash thereof), because when syncing from multiple sources, IDs may collide.
+		remoteLocalRef := resourceType + "/" + resource["id"].(string)
+		localResourceID := remoteRefToLocalRefMap[remoteLocalRef]
+		if localResourceID == "" {
+			localResourceID = generateLocalID()
+			remoteRefToLocalRefMap[remoteLocalRef] = localResourceID
+		}
```

Comment:
The `localResourceID` contains `urn:uuid:<UUID>`, which is a "bundle reference". When this bundle is POST'ed, the server will replace that bundle ref with the "actual" local id. So we'd need to set this reference from the response bundle, instead of this value I'd think?

Reply:
Ahhh you're absolutely right! I assumed it was a "global" in-memory map. Never mind my comment then!